docs/dynamictypes.rst

```diff
@@ -489,9 +544,46 @@ taking :class:`XXX` as our IDL type. These files provide a small Type Factory fo
 Generally, these files are not used directly, as now the type :class:`XXX` will register itself through its factory to
 ``TypeObjectFactory`` in its constructor, making very easy the use of static types with dynamic types.
 
+.. _discovery-time-data-typing:
+
+Discovery-Time Data Typing
+^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+When using fastdds API, if a participant discovers an endpoint which sends a complete TypeObject or a simple
+TypeIdentifier describing a type that the participant doesn't knows, it will be called to its listener's
+method ``on_type_discovery`` with the TypeInformation provided, and a ``DynamicType_ptr`` ready to be used.
```

Comment:
```suggestion
TypeIdentifier describing a type that the participant doesn't know, it will call to its listener's
``on_type_discovery`` method with the TypeInformation provided, and a ``DynamicType_ptr`` ready to be used.
```